react/Viewer/OnlyOfficeViewer.jsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import Button from '../Button'
+import { FileDoctype } from '../proptypes'
+
+import { withViewerLocales } from './withViewerLocales'
+import NoViewer from './NoViewer'
+
+const OnlyOfficeViewer = ({ file, onlyOfficeOpener, t }) => {
+  return (
+    <NoViewer
+      file={file}
+      renderFallbackExtraContent={() => (
+        <Button
+          label={t('Viewer.openInOnlyOffice')}
+          onClick={() => onlyOfficeOpener(file)}
+        />
+      )}
+    />
+  )
+}
+
+OnlyOfficeViewer.propTypes = {
+  file: FileDoctype,
+  onlyOfficeOpener: PropTypes.func.isRequired
+}
+
+export default withViewerLocales(OnlyOfficeViewer)
```

Comment:
We should create an hook for that one day

Reply:
👍 : https://github.com/cozy/cozy-ui/issues/1804